tls/s2n_tls13_secrets.c

```diff
@@ -520,6 +520,35 @@ S2N_RESULT s2n_derive_resumption_master_secret(struct s2n_connection *conn)
     return S2N_RESULT_OK;
 }
 
+/**
+ *= https://tools.ietf.org/rfc/rfc8446#section-7.1
+ *#           |
+ *#           +-----> Derive-Secret(., "exp master",
+ *#           |                     ClientHello...server Finished)
+ *#           |                     = exporter_master_secret
+ */
+S2N_RESULT s2n_derive_exporter_master_secret(struct s2n_connection *conn, struct s2n_blob *secret)
+{
+    RESULT_ENSURE_REF(conn);
+    /* Secret derivation requires these fields to be non-null.  */
+    RESULT_ENSURE_REF(conn->secure);
+    RESULT_ENSURE_REF(conn->secure->cipher_suite);
+
+    RESULT_GUARD(s2n_derive_secret_with_context(conn,
+            S2N_MASTER_SECRET,
+            &s2n_tls13_label_exporter_master_secret,
+            SERVER_FINISHED,
+            secret));
+
+    if (conn->secret_cb && (s2n_connection_is_quic_enabled(conn) || s2n_in_unit_test())) {
```

Comment:
I really don't like the repetition of this check, and less strongly don't like that the callbacks can be triggered from two different places. Can we do some cleanup / refactoring to avoid that?